src/main/java/com/tabnine/lifecycle/BinaryPromotionStatusBarLifecycle.kt

```diff
@@ -1,61 +1,21 @@
 package com.tabnine.lifecycle
 
-import com.intellij.openapi.project.ProjectManager
-import com.intellij.openapi.wm.WindowManager
 import com.tabnine.binary.BinaryRequestFacade
-import com.tabnine.binary.requests.statusBar.StatusBarPromotionBinaryRequest
-import com.tabnine.binary.requests.statusBar.StatusBarPromotionShownRequest
 import com.tabnine.general.StaticConfig.BINARY_PROMOTION_POLLING_DELAY
 import com.tabnine.general.StaticConfig.BINARY_PROMOTION_POLLING_INTERVAL
-import com.tabnine.statusBar.StatusBarPromotionWidget
+import com.tabnine.statusBar.StatusBarUpdater
 import java.util.Timer
-import java.util.TimerTask
+import kotlin.concurrent.timerTask
 
 class BinaryPromotionStatusBarLifecycle(private val binaryRequestFacade: BinaryRequestFacade) {
+    private val statusBarUpdater = StatusBarUpdater(binaryRequestFacade)
```

Comment:
Get it in the ctor instead of creating it

Reply:
Done (though I think this is over encapsulation).